packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -1146,4 +1053,283 @@ void main() {
     expect(find.text('hello'), findsOneWidget);
     expect(called, 1);
   });
+
+  group('SnackBar position', () {
+    for (final SnackBarBehavior behavior in SnackBarBehavior.values) {
+      final SnackBar snackBar = SnackBar(
+        content: const Text('SnackBar text'),
+        behavior: behavior,
+      );
+
+      testWidgets(
+        '$behavior should align SnackBar with the bottom of Scaffold '
+        'when Scaffold has no other elements',
+        (WidgetTester tester) async {
+          await tester.pumpWidget(
+            MaterialApp(
+              home: Scaffold(
+                body: Container(),
+              ),
+            ),
+          );
+
+          final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+          scaffoldState.showSnackBar(snackBar);
+
+          await tester.pump();
+
+          final Offset snackBarBottomRight = tester.getBottomRight(find.byType(SnackBar));
+          final Offset scaffoldBottomRight = tester.getBottomRight(find.byType(Scaffold));
+
+          expect(snackBarBottomRight, equals(scaffoldBottomRight));
+
+          final Offset snackBarBottomLeft = tester.getBottomLeft(find.byType(SnackBar));
+          final Offset scaffoldBottomLeft = tester.getBottomLeft(find.byType(Scaffold));
+
+          expect(snackBarBottomLeft, equals(scaffoldBottomLeft));
+        },
+      );
+
+      testWidgets(
+        '$behavior should align SnackBar with the top of BottomNavigationBar '
+        'when Scaffold has no FloatingActionButton',
+        (WidgetTester tester) async {
+          final UniqueKey boxKey = UniqueKey();
+          await tester.pumpWidget(
+            MaterialApp(
+              home: Scaffold(
+                body: Container(),
+                bottomNavigationBar: SizedBox(key: boxKey, width: 800, height: 60),
+              ),
+            ),
+          );
+
+          final ScaffoldState scaffoldState = tester.state(find.byType(Scaffold));
+          scaffoldState.showSnackBar(snackBar);
+
+          await tester.pump();
```

Comment:
You might want the SnackBar to fully animate out before checking the positions.
```suggestion
          await tester.pumpAndSettle(); // Have the SnackBar fully animate out.
```

Reply:
Fixed.